nUpdate.UpdateInstaller/UI/Popups/Popup.cs

```diff
@@ -62,6 +72,10 @@ public static DialogResult ShowPopup(Icon popupIcon, string title, Exception ex,
         public static DialogResult ShowPopup(IWin32Window owner, Icon popupIcon, string title, string infoMessage,
             PopupButtons buttons)
         {
+            if (LoggedIntoEventlog(popupIcon, infoMessage))

```

Comment:
What is the sense of that? This would suppress any popups and log the messages instead, wouldn't it? But, this is not what should happen when we want to see the messages inside a popup message. Or am I missing something?

Reply:
no, would not
if the application runs as service, you cant open dialogs- so we need to write in to the eventlog - without that we wouldn't see any messages/exceptions from nUpdate

you can open the window if "Environment.UserInteractive" is true, but in a terminel server environment every user would get a confusing popup. so i decide this would be the best solution

Reply:
The thing I think about is: Once we have an app runing as service, we do not initialize the usual reporter service with the integrated user interface (https://github.com/ProgTrade/nUpdate/pull/62/commits/956f956123bedcc3690318826a4695c916ad3b43#diff-b0b1bcab7f1b1a86e0c87a87fa9d91deR32). But, the MainForm is the only class that calls popups. So, isn't this call redundant then? When we do not show the MainForm at all in case of a service, we do not have other possibilites to create a popup in the update installer. Or am I missing a call? Or is this a general safety feature in case the popups may be used (accidentally)?

Reply:
[Program.cs#L158](https://github.com/ProgTrade/nUpdate/blob/master/nUpdate.UpdateInstaller/Program.cs#L158)
[Program.cs#L199](https://github.com/ProgTrade/nUpdate/blob/master/nUpdate.UpdateInstaller/Program.cs#L199)
[Updater.cs#L126](https://github.com/ProgTrade/nUpdate/blob/master/nUpdate.UpdateInstaller/Updater.cs#L126)

Reply:
Okay, thanks. I don't know why I didn't remember/see these calls there. Seems to be a long time that I've written the code. =)